sourcegraph/jetbrains
# Add multi-repo enterprise context

## Test plan

1. Open `sourcegraph/cody` project with non-enterprise account.
- Open new chat and ask question about current repo (e.g. some class) - assistant should know the answer
- Open new chat and ask question about squirrel - assistant should describe you an animal
- Open new chat and disable local context. Ask about current repo (e.g. some class) - assistant should not have a context
- Save current context as default. Close the IDE. Reopen the IDE. 
  - Go to Chat History tab and open previous chats one by one. Check if both history and context settings are properly preserved.
  - Open new chat and check if context is disabled. It should be, as we previously set that as new default. Enable it again and set as default.
  
2. Open `sourcegraph/cody` project with enterprise account.
- All checks from point 1) apply there as well.
- Click [+] button in the context panel and type sourcegraph repo url (`github.com/sourcegraph/sourcegraph`)
  - Check if validator blocks from entering incomplete or invalid URL
  - Add the `sourcegraph/sourcegraph` repo by hitting Add button
- Open new chat and ask question about squirrel - assistant should describe you an HTTP server, **NOT** animal.
- Set new configuration as default and open new chat. It should use previous context configuration.
- Disable `sourcegraph/sourcegraph` remote repo context.
- Ask question about squirrel. It should be back to describing you an animal or having no context.
- Save current context as default. Close the IDE. Reopen the IDE. 
  - Go to Chat History tab and open previous chats one by one. Check if both history and context settings are properly preserved.
  - Open new chat and check if `sourcegraph/sourcegraph` is disabled. It should be, as we previously set that as new default. Enable it again and set as default.
  - Open new chat and check if `sourcegraph/sourcegraph` is enabled, it should be.
  - Remove `sourcegraph/sourcegraph` repo by clicking on it to select it, and then clicking [-] button.
  - Ask question about squirrel. It should be back to describing you an animal or having no context.

## Changes

* Added multi-repo enterprise context
* Improved look of the local context panel
* Fixed several bugs like:
  * history service being instantiated as application-wide service instead of project-wide
  * current repo url parsing and timing issues
 
![image](https://github.com/sourcegraph/jetbrains/assets/1519649/91a014db-6a64-4418-aee6-96400c25374f)
